sdk/dotnet/Pulumi/Deployment/Deployment.Runner.cs

```diff
@@ -41,9 +57,11 @@ public Task<int> RunAsync(Func<Task<IDictionary<string, object?>>> func)
 
             public void RegisterTask(string description, Task task)
             {
-                lock (_taskToDescription)
+                Serilog.Log.Information($"Registering task: {description}");
+
+                lock (_inFlightTasks)
                 {
-                    _taskToDescription.Add(task, description);
+                    _inFlightTasks.AddLast((task, description));
```

Comment:
Why do we need to add duplicate tasks to the list instead of ignoring them? Maybe leave a comment in the code of why we could use a dictionary here.

Reply:
Edited with "could not"

Reply:
I like the symettry of both logging that we got the task and logging that we completed it.  alternatively, we could have the dcitionary map to multiple descriptions.  Do you have a preference?

Reply:
A dictionary would look cleaner to me

Reply:
Done!